### PR TITLE
Add integration test for cross-mounting blobs

### PR DESCRIPTION
**Pull request by dmage**
/assign @legionus

### CODE REVIEW

**Comment by legionus** [comment]
Hm ...
```go
if s := os.Getenv("REGISTRY_HTTP_TLS_MINVERSION"); len(s) > 0 {
```
```go
if s := os.Getenv("REGISTRY_HTTP_TLS_CIPHERSUITES"); len(s) > 0 {
```
There are no such fields in the configuration [1][2]. It's unclear why it was necessary to name these environment variables like that.

I think it will be better to move them to our section in the configuration file. I don't mean in this PR.

[1] https://docs.docker.com/registry/configuration/#http
[2] https://github.com/openshift/image-registry/blob/master/vendor/github.com/docker/distribution/configuration/configuration.go#L91

File: pkg/cmd/dockerregistry/dockerregistry.go [hunk 1]
```diff
@@ -157,20 +158,47 @@ func Execute(configFile io.Reader) {
...
 
-// Start runs the Docker registry. Start always returns a non-nil error.
-func Start(ctx context.Context, dockerConfig *configuration.Configuration, extraConfig *registryconfig.Configuration) error {
-	setDefaultLogParameters(dockerConfig)
+// tcpKeepAliveListener sets TCP keep-alive timeouts on accepted
+// connections. It's used by ListenAndServe and ListenAndServeTLS so
+// dead TCP connections (e.g. closing laptop mid-download) eventually
+// go away.
+type tcpKeepAliveListener struct {
+	*net.TCPListener
```
[the remark this answers is not in the document]
Comment (dmage) [reply pkg/cmd/dockerregistry/dockerregistry.go]: No, it is an embedded field on purpose.

**Pull request merged**